modin/pandas/groupby.py

```diff
@@ -224,6 +227,13 @@ def _shift(data, periods, freq, axis, fill_value, is_set_nan_rows=True):
 
         if freq is None and axis == 1 and self._axis == 0:
             result = _shift(self._df, periods, freq, axis, fill_value)
+
+            # This workaround is connected with pandas GH 44269 to align a row order with pandas.
+            import pandas.core.algorithms as algorithms
+
+            indexer, _ = result.index.get_indexer_non_unique(result.index._values)
+            indexer = algorithms.unique1d(indexer)
+            result = result.take(indexer)
```

Comment:
If this is a bug we shouldn't math its behaviour. I suggest removing this logic and adding an extra sort at the `groupby.shift` test with the comment of why we're doing it.

Reply:
moved to `eval_shift` test function